block_iagora.php

```diff
@@ -66,12 +106,18 @@ public function get_content() {
      * @return string The generated HTML content for the chat.
      */
     private function generate_chat_content($copilotendpointurl) {
-        global $OUTPUT;
+        global $OUTPUT, $USER, $COURSE;
         // Generate a unique identifier for the chat container.
         $chatid = uniqid('iagora_chat_');
+
+    // Si la complétion est activée et que l'utilisateur visualise l'activité 
+    if ($completion->is_enabled($cm) && $completion->is_tracked_user($USER->id)) {
+        $completion->set_module_viewed($cm);  // Marquer l'activité comme vue
+    }
         $context = [
             'chatId' => $chatid,
             'tokenEndpointURL' => $copilotendpointurl,
+            'progress' => $progressPercentage
```

Comment:
```suggestion
            'progressPercentage' => $progressPercentage
```
The key should match the one used in the Mustache template.